iproto: make iproto resistant to misusage (#10589)
tarantool/tarantool: changelogs/unreleased/gh_10155_make_iproto_resistant_to_misusage.md

> ## bugfix/iproto
> 
> * Fixed an issue when Tarantool could easily crash if iproto was used
>   incorrectly, not according to the protocol (gh-10155).

Reviewer: - We never used "iproto" section. Better use "core" for this bug.
- I'd be more specific about what happens here.

```suggestion
## bugfix/core

* Fixed a bug when a server could crash if a client sent an IPROTO replication
  request without waiting for pending requests to complete (gh-10155).
```

Author: Fixed.

> * Fixed an issue when the tarantool could be easily crashed using iproto
>   incorrectly, not according to the protocol (gh-10155).

Reviewer 2: ```suggestion
* Fixed an issue when Tarantool could crash if iproto was used incorrectly, not according to the protocol (gh-10155).
```
